File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/04.For-Loop/02-exercise/03.histogram.js
function histogram(input) {

    let index = 0;
    let num = Number(input[index]);
    index++;

    let p1Counter = 0;
    let p2Counter = 0;
    let p3Counter = 0;
    let p4Counter = 0;
    let p5Counter = 0;

    for (let i = 0; i < num; i++) {
        let x = Number(input[index]);
        index++;

        if (x < 200) {
            p1Counter += 1;
        } else if (x >= 200 && x < 400) {
            p2Counter += 1;
        } else if (x >= 400 && x < 600) {
            p3Counter += 1;
        } else if (x >= 600 && x < 800) {
            p4Counter += 1;
        } else if (x >= 800) {
            p5Counter += 1;
        }
    }

    const p1Percentage = p1Counter / num * 100;
    const p2Percentage = p2Counter / num * 100;
    const p3Percentage = p3Counter / num * 100;
    const p4Percentage = p4Counter / num * 100;
    const p5Percentage = p5Counter / num * 100;

    console.log(p1Percentage.toFixed(2) + "%");
    console.log(p2Percentage.toFixed(2) + "%");
    console.log(p3Percentage.toFixed(2) + "%");
    console.log(p4Percentage.toFixed(2) + "%");
    console.log(p5Percentage.toFixed(2) + "%");
}
histogram(["9", "367", "99", "200", "799", "999", "333", "555", "111", "9"]);

File: Python_Basics/Homeworks-And-Labs/04.for_loop/02.exercise/03.histogram.py
num = int(input())

p1_counter = 0
p2_counter = 0
p3_counter = 0
p4_counter = 0
p5_counter = 0

for i in range(num):
    current = int(input())

    if current < 200:
        p1_counter += 1
    elif current >= 200 and current < 400:
        p2_counter += 1
    elif current >= 400 and current < 600:
        p3_counter += 1
    elif current >= 600 and current < 800:
        p4_counter += 1
    elif current >= 800:
        p5_counter += 1

p1_percentage = p1_counter / num * 100
p2_percentage = p2_counter / num * 100
p3_percentage = p3_counter / num * 100
p4_percentage = p4_counter / num * 100
p5_percentage = p5_counter / num * 100

print(f'{p1_percentage:.2f}%')
print(f'{p2_percentage:.2f}%')
print(f'{p3_percentage:.2f}%')
print(f'{p4_percentage:.2f}%')
print(f'{p5_percentage:.2f}%')
